models/step_run_status.go

```diff
@@ -0,0 +1,79 @@
+package models
+
+// StepRunStatus ...
+type StepRunStatus int
+
+const (
+	StepRunStatusCodeSuccess StepRunStatus = iota
+	StepRunStatusCodeFailed
+	StepRunStatusCodeFailedSkippable
+	StepRunStatusCodeSkipped
+	StepRunStatusCodeSkippedWithRunIf
+	StepRunStatusCodePreparationFailed
+	StepRunStatusAbortedWithCustomTimeout   // step times out due to a custom timeout
+	StepRunStatusAbortedWithNoOutputTimeout // step times out due to no output received (hang)
+)
```

Comment:
I want to highlight that in the original constant list the value 6 was missing. So `StepRunStatusCodePreparationFailed` was 5 and `StepRunStatusAbortedWithCustomTimeout` 7.  With this change the last two statuses will not have their original value.

And this value reordering and using the iota might be dangerous ot use. If someone inserts a new state in the middle of the list (or reintroduces the missing 6 value) then the whole status is off. And this is used by (based on GitHub code search):
- build log service
- bitrise analytics plugin
and based on the code they use it for mapping purposes. 

Either lets use dedicated integer values like before or add a big comment to warn not to modify the order of the list because other tools might depend on them.

Reply:
Thanks for spotting this, it was not intentional, restored here: https://github.com/bitrise-io/bitrise/pull/833/commits/b58b743081a03afee8ca9171cc390688648e9563